runtimeconfig/google/cloud/runtimeconfig/variable.py

```diff
@@ -0,0 +1,197 @@
+        :returns: If set, one of "UPDATED", "DELETED", or
+                  "VARIABLE_STATE_UNSPECIFIED", else ``None``.
+        """
+        return self._properties.get('state')
+
+    @property
+    def update_time(self):
+        """Retrieve the timestamp at which the variable was updated.
+
+        See:
+        https://cloud.google.com/deployment-manager/runtime-configurator/reference/rest/v1beta1/projects.configs.variables
+
+        :rtype: :class:`datetime.datetime` or ``NoneType``
+        :returns: Datetime object parsed from RFC3339 valid timestamp, or
+                  ``None`` if the property is not set locally.
+        """
+        value = self._properties.get('updateTime')
+        if value:
+            value = _rfc3339_to_datetime(value)
+        return value
+
+    def _require_client(self, client):
+        """Check client or verify over-ride.
+
+        :type client: :class:`google.cloud.runtimconfig.client.Client` or
+                      ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current zone.
+
+        :rtype: :class:`google.cloud.runtimeconfig.client.Client`
+        :returns: The client passed in or the currently bound client.
+        """
+        if client is None:
+            client = self.client
+        return client
+
+    def _set_properties(self, api_response):
+        """Update properties from resource in body of ``api_response``
+
+        :type api_response: httplib2.Response
+        :param api_response: response returned from an API call
+        """
+        self._properties.clear()
+        cleaned = api_response.copy()
+        if 'name' in cleaned:
+            self.name = variable_name_from_full_name(cleaned.pop('name'))
+        self._properties.update(cleaned)
+
+    def exists(self, client=None):
+        """API call:  test for the existence of the variable via a GET request
+
+        See:
+        https://cloud.google.com/deployment-manager/runtime-configurator/reference/rest/v1beta1/projects.configs.variables/get
+
+        :type client: :class:`~google.cloud.runtimeconfig.client.Client` or
+                      ``NoneType``
+        :param client: Optional. The client to use.  If not passed, falls back
+                       to the ``client`` stored on the variable's config.
+
+        :rtype: boolean
+        :returns: True if the variable exists in Cloud RuntimeConfig.
+        """
+        client = self._require_client(client)
+        try:
+            # We only need the status code (200 or not) so we seek to
+            # minimize the returned payload.
+            query_params = {'fields': 'name'}
+            client.connection.api_request(method='GET', path=self.path,
+                                          query_params=query_params)
+            return True
+        except NotFound:
+            return False
+
+    def reload(self, client=None, fields=None):
```

Comment:
Since I was copying the storage sample, I got the fields behavior from here:
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/a7cb215d1fb54e00299778d5d04f55893a5f0c9f/storage/google/cloud/storage/bucket.py#L271

and here:
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/5e3a35e63474e56df39f906e71a2746f390f5b5e/storage/google/cloud/storage/client.py#L246

Do we have an example of not passing it through directly?

(The reason I added this parameter is I didn't know how to cover the branches where the variable properties are not set in a way that makes sense. Without passing the fields value to the API, and updateTime will always be present, for example)